### PR TITLE
feat: standardize to use CPU for pooling

### PR DESCRIPTION
**Pull request by junjie-landing**

### CODE REVIEW

File: vision_agent_tools/models/florencev2.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def _process_image(self, image: Image.Image) -> Image.Image:
     def _process_video(self, images: VideoNumpy) -> list[Image.Image]:
         return [self._process_image(Image.fromarray(arr)) for arr in images]
 
-    def __init__(self):
+    def __init__(self, device: Optional[Device] = Device.GPU) -> None:
```
Comment (hrnn) [inline vision_agent_tools/models/florencev2.py:65]: nit: we can use modern syntax for typing 
```suggestion
    def __init__(self, device: Device | None = Device.GPU) -> None:
```

**Review by hrnn — APPROVED** [review]
:shipit: LGTM

File: docs/shared_model_manager.md [hunk 2]
```diff
@@ -11,13 +11,15 @@ Models are only loaded into memory when needed using the `fetch_model` function.
 
 The usage example demonstrates adding models and then using them with their respective functionalities.
 
+⚠️ ❕: we should ALWAYS **add model instance to CPU** , and model pool will automatically put it in GPU.
```
Comment (Dayof) [inline docs/shared_model_manager.md:14]: didn't quite understand this phrase, if we set to CPU why it will be automatically put it GPU?

Comment (junjie-landing) [reply docs/shared_model_manager.md:14]: Lemme explain more in doc

**Review by Dayof — APPROVED** [review]

**Pull request merged**